src/test/java/org/openrefine/extensions/commons/utils/CommonsImportingControllerTest.java

```diff
@@ -0,0 +1,210 @@
+            }
+        }
+        return project2;
+    }
+
+    @BeforeMethod
+    public void setUp() throws IOException, ModelException {
+
+        MockitoAnnotations.initMocks(this);
+
+        File dir = createTempDirectory("OR_CommonsExtension_Test_WorkspaceDir");
+        FileProjectManager.initialize(dir);
+
+        servlet = new RefineServlet();
+        ImportingManager.initialize(servlet);
+        project = new Project();
+        metadata = new ProjectMetadata();
+
+        metadata.setName("Commons Import Test Project");
+        ProjectManager.singleton.registerProject(project, metadata);
+        SUT = new CommonsImportingController();
+
+        project2 = createProjectWithColumns("UrlFetchingTests", "files");
+
+    }
+
+    @AfterMethod
+    public void tearDown() {
+        SUT = null;
+        request = null;
+        response = null;
+        project = null;
+        metadata = null;
+        job = null;
+    }
+
+    @Test
+    public void testDoGet() {
+        StringWriter sw = new StringWriter();
+        PrintWriter pw = new PrintWriter(sw);
+
+        try {
+            when(response.getWriter()).thenReturn(pw);
+
+            SUT.doGet(request, response);
+            System.out.print(sw +"1\n");
+
+            String result = sw.getBuffer().toString().trim();
+            System.out.print(result +"2\n");
+            System.out.print(pw +"3\n");
+            ObjectNode json = ParsingUtilities.mapper.readValue(result, ObjectNode.class);
+            String code = json.get("status").asText();
+            String message = json.get("message").asText();
+            Assert.assertNotNull(code);
+            Assert.assertNotNull(message);
+            Assert.assertEquals(code, "error");
+            Assert.assertEquals(message, "GET not implemented");
+
+        } catch (Exception e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Test for caching
+     */
+    @Test
+    public void testUrlCaching() throws Exception {
+        try (MockWebServer server = new MockWebServer()) {
+            server.start();
+            HttpUrl url = server.url("/w/api.php");
+
+            Row row = new Row(2);
+            row.setCell(0, new Cell("File:\"Amit Trainin drawing mural painting at Adam Zair magazine office - July 2021.jpg\".jpg", null));
+            project2.rows.add(row);
+            server.enqueue(new MockResponse().setBody(Integer.toString(100)));
+            String data = project2.rows.get(0).getCellValue(0).toString();
+
+            EngineDependentOperation op = new ColumnAdditionByFetchingURLsOperation(engine_config,
+                    "files",
+                    url + "?action=query&list=categorymembers&cmtitle=Category:art&cmtype=file&cmlimit=5&format=json",
+                    OnError.StoreError,
+                    "new",
+                    1,
+                    500,
+                    true,
+                    null);
```

Comment:
This test is testing the `ColumnAdditionByFetchingURLsOperation`, which has nothing to do with the Commons extension, so we should remove it and instead have some tests which test the fetching of Commons categories via your ImportingController.